sklearn/tree/_tree.pyx

```diff
@@ -253,16 +262,35 @@ cdef class DepthFirstTreeBuilder(TreeBuilder):
                 # inspection and interpretation
                 splitter.node_value(tree.value + node_id * tree.value_stride)
 
+                middle_value = tree.value[node_id]
```

Comment:
That's not the "middle" value, is it? IIRC there were issues to know the sibling of a node?

Reply:
The middle value only has to be between the left child's value and the the right child's value. I took the parent's value as it is guaranteed to be there and readily available, unlike the average of the children's value.